tests/integration/renderer/test_request_hooks.py

```diff
@@ -295,3 +295,46 @@ def wrap(*args, **kwargs):
     dash_duo.wait_for_text_to_equal("#output-token", "..")
 
     assert len(dash_duo.get_logs()) == 2
+
+
+def test_rdrh004_layout_hooks(dash_duo):
+    app = Dash(__name__)
+
+    app.index_string = """<!DOCTYPE html>
+    <html>
+        <head>
+            {%metas%}
+            <title>{%title%}</title>
+            {%favicon%}
+            {%css%}
+        </head>
+        <body>
+            {%app_entry%}
+            <footer>
+                {%config%}
+                {%scripts%}
+                <script id="_dash-renderer" type"application/json">
+                    const renderer = new DashRenderer({
```

Comment:
This api usage is not something I think we should expand upon without improvements, it's really hard to use and has boilerplate not required for the feature.

I suggest we add parameter(s) that get them in the main index without the need to override the index string.

Reply:
oh sure, I was just following the same style as the other tests in this file.
I agree it's odd to overwrite the entire index_string. I can also do this instead:

```
app.renderer = """
    new DashRenderer({
        ... 
    });
"""
```

which would resemble what I'm actually doing in EnterpriseDash

Reply:
No I meant we should add a parameter to init like:
```python
app = Dash(hooks={"pre_layout": "() => null"})
```
This way you don't need to override the index values and the boilerplate is minimal.

Reply:
As discussed, I've added the ability to pass these hooks as an argument to the constructor.